### ISSUE TITLE
how did it work in sample without @EnableDubbo?

### ISSUE THREAD

**Issue by lfz757077613** [apache/incubator-dubbo-spring-boot-project#286]
I use the 0.2.0 version, but fail to run the demo copied from the sample
Finally, I find It works well after adding @EnableDubbo on BootdubboProviderApplication
I guess The sample has not bean updated after dubbo-2.5.8, because @EnableDubbo was add at dubbo-2.5.8.
At last, There isn't enough wiki for the real environment such as zk registry in dubbo-spring-boot. It cast the whole day to config registry center on me, a practised user with springmvc and dubbo

**Comment by lfz757077613** [comment]
#### After reading source code and testing, I located the cause.
- if we set dubbo properties in yml or properties, DubboAutoConfiguration will work. So actually @EnableDubbo is needless, it just effect DubboComponentScan
- If we set dubbo.scan.basePackages in properties or @EnableDubbo(scanBasePackages = "xx"), the classes annotated by dubbo @Service will be managed by spring. But if we use yml and set "dubbo: scan: base-packages: xx" following the wiki, a error will be thrown. Because even if we use yml, finally the property is also got from "basePackages", so those classes annotated by dubbo @Service will not be managed by spring meaning no privider error
Thus, we can set this as below in yml or use properties or @EnableDubbo(scanBasePackages = "xx") to set basePackages 

```
dubbo:
  scan:
     basePackages : xx
```

- PS:  If using yml or properties, there are warn log in console as below. If using @EnableDubbo(scanBasePackages = "xx"), there is only first warn log shown.
And I wonder why

```
2018-09-23 19:09:06.153  WARN 6348 --- [           main] b.f.a.ServiceAnnotationBeanPostProcessor :  [DUBBO] The BeanDefinition[Root bean: class [com.alibaba.dubbo.config.spring.ServiceBean]; scope=; abstract=false; lazyInit=false; autowireMode=0; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=null; factoryMethodName=null; initMethodName=null; destroyMethodName=null] of ServiceBean has been registered with name : ServiceBean:defaultDemoService:com.alibaba.boot.dubbo.demo.consumer.DemoService:${demo.service.version}, dubbo version: 2.6.2, current host: 192.168.199.223

2018-09-23 19:09:06.154  WARN 6348 --- [           main] o.s.c.a.ConfigurationClassPostProcessor  : Cannot enhance @Configuration bean definition 'com.alibaba.boot.dubbo.autoconfigure.DubboAutoConfiguration' since its singleton instance has been created too early. The typical cause is a non-static @Bean method with a BeanDefinitionRegistryPostProcessor return type: Consider declaring such methods as 'static'.

```

**Comment by mercyblitz** [comment]
That's an issue about YAML file support, let me check and fix it

**Comment by mercyblitz** [comment]
It's fixed on Dubbo 2.6.5, you could update the dependency first.
